### PR TITLE
New test cases for migrations

### PR DESCRIPTION
**Pull request by leabstrait**
Addresses issues #25 and #26 

The order of migration and rollback test commands are:

1. Create table and migrate all in folder
2. Migrate Empty
3. Rollback * 2
4. Migrate (2)
5. Rollback * 3
6. Rollback empty

I had to change the sequence to be different from what @halvardssm proposed because I couldn't find a way to add new migration scripts in midst of the running tests.

### CODE REVIEW

File: tests/migration.test.ts [hunk 1]
```diff
@@ -24,17 +26,41 @@ const strings = [
   {
     name: "Rollback",
```
Comment (halvardssm) [inline tests/migration.test.ts:27]:
```suggestion
    name: "Rollback Mango",
```

File: tests/migration.test.ts [hunk 2]
```diff
@@ -24,17 +26,41 @@ const strings = [
...
     string: TYPE_ROLLBACK,
-    solution: ["Rolled back 1587937822648-test.ts"],
+    solution: ["Rolled back 1587937822650-mango.ts"],
   },
   {
-    name: "Rollback empty",
+    name: "Rollback",
```
Comment (halvardssm) [inline tests/migration.test.ts:32]:
```suggestion
    name: "Rollback Apple",
```

File: tests/migration.test.ts [hunk 3]
```diff
@@ -24,17 +26,41 @@ const strings = [
...
     string: TYPE_ROLLBACK,
-    solution: ["Nothing to rollback"],
+    solution: ["Rolled back 1587937822649-apple.ts"],
   },
   {
     name: "Migrate",
```
Comment (halvardssm) [inline tests/migration.test.ts:37]:
```suggestion
    name: "Migrate Apple and Mango",
```

File: tests/migration.test.ts [hunk 4]
```diff
@@ -24,17 +26,41 @@ const strings = [
...
     string: TYPE_MIGRATE,
-    solution: ["Migrated 1587937822648-test.ts", "Migration complete"],
+    solution: [
+      "Migrated 1587937822649-apple.ts",
+      "Migrated 1587937822650-mango.ts",
+      "Migration complete",
+    ],
+  },
+  {
+    name: "Rollback",
```
Comment (halvardssm) [inline tests/migration.test.ts:46]:
```suggestion
    name: "Rollback Mango",
```

File: tests/migration.test.ts [hunk 5]
```diff
@@ -24,17 +26,41 @@ const strings = [
...
+    string: TYPE_ROLLBACK,
+    solution: ["Rolled back 1587937822650-mango.ts"],
+  },
+  {
+    name: "Rollback",
```
Comment (halvardssm) [inline tests/migration.test.ts:51]:
```suggestion
    name: "Rollback Apple",
```

File: tests/migration.test.ts [hunk 6]
```diff
@@ -24,17 +26,41 @@ const strings = [
...
+    string: TYPE_ROLLBACK,
+    solution: ["Rolled back 1587937822649-apple.ts"],
+  },
+  {
+    name: "Rollback",
```
Comment (halvardssm) [inline tests/migration.test.ts:56]:
```suggestion
    name: "Rollback Test",
```

**Comment by halvardssm** [comment]
@leabstrait You can press "Update branch" to get the latest changes from master into this branch 👍

**Comment by leabstrait** [comment]
MYSQL and SQLite are failing. I had only Postgres in my machine so only ran the tests through them.

**Comment by halvardssm** [comment]
Screenshot 2020-05-23 at 20 05 33

So the issue regarding the failing test lays somewhere in the filter and sort function 🤔 

Do you have Docker? If you do, you can simply run `make test-clean` and it will boot up all the databases in docker and run the tests.

File: cli/utils.ts [hunk 7]
```diff
@@ -42,14 +42,14 @@ export const filterAndSortFiles = (
...
+    .filter((file: Deno.DirEntry): boolean => {
+      if (!regexFileName.test(file.name)) return false;
 
-    if (queryResult === undefined) return true;
+      if (queryResult === undefined) return true;
 
-    return parseInt(file.name.split("-")[0]) >
-      new Date(queryResult).getTime();
-  })
+      return file.name > queryResult[0];
```
Comment (halvardssm) [inline cli/utils.ts:51]:
```suggestion
      return file.name > queryResult;
```

Comment (leabstrait) [reply cli/utils.ts:51]: Is queryResult an array or a string? In PostgreSQL case it used to return an array hence the array index of 0. Should I commit this just to check?

Comment (halvardssm) [reply cli/utils.ts:51]: The query result will always be a string or undefined (see the type), the logic for parsing the arrays are done before this method is called

**Comment by leabstrait** [comment]
Okay, It's good to go I guess. Thanks for the help. Next stop, learn docker 😁

**Comment by halvardssm** [comment]
LGTM! Thanks for your contribution! 💪 Regarding Docker, it is really useful for testing and development. Absolutely worth learning it.

**Pull request merged**